### ISSUE TITLE
No value for foods

### ISSUE THREAD

**Issue by uzair157** [EugeneHoran/Android-FatSecret-REST-API#3]
I implemented your code and there were no errors. I run it on my device but when i search an item in the search bar, on the logcat of eclipse I recieve error "No value for foods".
Please help me I am stuck on this issue for past 3 days.

**Comment by EugeneHoran** [comment]
Did you sign up for their api key for your app?

**Comment by uzair157** [comment]
Yes i did get the api key as well as the secret key, still no luck !

I tried copying the project files on another eclipse set up but get following errors in style.xml
it states "Error: No resource found that matches the given name: attr 'colorPrimaryDark'.".
Please help me, I need this api working for my Final year project.

Thanks alot for responding!

**Comment by uzair157** [comment]
![error](https://cloud.githubusercontent.com/assets/13968441/9500364/6eb4e7b0-4c3e-11e5-8c1d-589fb1106846.JPG)

**Comment by uzair157** [comment]
I have identified the function that is causing error. the error is caused by the function shown in the pic below. "foods = food.getJSONObject("food");" is throwing the exception. Please help me with this problem i will be very thankful.
![errorpic](https://cloud.githubusercontent.com/assets/13968441/9557292/4cdc6c32-4df4-11e5-9ed4-9dd0b78dd4b9.JPG)

**Comment by siddhantjain** [comment]
this might be a very delayed comment, but if the code you posted above is anything to go by, the problem with it is in the line:
foods = good.getJSONObject("food");
The string should be "foods". Hence, the code becomes:
foods = good.getJSONObject("foods");

**Comment by andreluizreis** [comment]
2016, I am in the same problem @uzair157. 
I found that it is a problem with the authentication. 
{ "error": {"code": 8, "message": "Invalid signature: oauth_signature 'YcFEXTtpJHILCdjtqAKgskB2WP4='" }}
I don't know if something on the API changed. 
@EugeneHoran, can you confirm if your code is running correctly with your credentials  yet?

**Comment by andreluizreis** [comment]
@uzair157, @EugeneHoran, I saw now the other post about the "&" at the end of the secret password. 
Thank you. It is now working.
